File: upl-purchase-sdk-js/src/unzer-paylater/model/PurchaseInformation.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

const ModelHelper = require("../util/ModelHelper");
const Amount = require("./Amount");
const Consumer = require("./Consumer");
const ConsumerVerification = require("./ConsumerVerification");
const Currency = require("./Currency");
const Document = require("./Document");
const MerchantReference = require("./MerchantReference");
const OperationInformation = require("./OperationInformation");
const PaymentInformation = require("./PaymentInformation");
const PaymentOption = require("./PaymentOption");
const PurchaseState = require("./PurchaseState");

/**
 * Describes the current state of a purchase.
 */
class PurchaseInformation {
    /**
     * @returns { String }
     */
    getPurchaseId() {
        return this.purchaseId;
    }
    /**
     * @param { String } purchaseId
     */
    setPurchaseId(purchaseId) {
        this.purchaseId = ModelHelper.validatePrimitive(purchaseId, "string");
    }
    /**
     * @param { String } val
     */
    withPurchaseId(val) {
        this.setPurchaseId(val);
        return this;
    }

    /**
     * @returns { PurchaseState }
     */
    getState() {
        return this.state;
    }
    /**
     * @param { PurchaseState } state
     */
    setState(state) {
        this.state = ModelHelper.validateEnum(state, PurchaseState, "PurchaseState");
    }
    /**
     * @param { PurchaseState } val
     */
    withState(val) {
        this.setState(val);
        return this;
    }

    /**
     * @returns { Currency }
     */
    getCurrency() {
        return this.currency;
    }
    /**
     * @param { Currency } currency
     */
    setCurrency(currency) {
        this.currency = ModelHelper.validateEnum(currency, Currency, "Currency");
    }
    /**
     * @param { Currency } val
     */
    withCurrency(val) {
        this.setCurrency(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getAuthorizedAmount() {
        return this.authorizedAmount;
    }
    /**
     * @param { Amount } authorizedAmount
     */
    setAuthorizedAmount(authorizedAmount) {
        this.authorizedAmount = ModelHelper.validateObject(authorizedAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withAuthorizedAmount(val) {
        this.setAuthorizedAmount(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getCapturedAmount() {
        return this.capturedAmount;
    }
    /**
     * @param { Amount } capturedAmount
     */
    setCapturedAmount(capturedAmount) {
        this.capturedAmount = ModelHelper.validateObject(capturedAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withCapturedAmount(val) {
        this.setCapturedAmount(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getRemainingCaptureAmount() {
        return this.remainingCaptureAmount;
    }
    /**
     * @param { Amount } remainingCaptureAmount
     */
    setRemainingCaptureAmount(remainingCaptureAmount) {
        this.remainingCaptureAmount = ModelHelper.validateObject(remainingCaptureAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withRemainingCaptureAmount(val) {
        this.setRemainingCaptureAmount(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getRefundedAmount() {
        return this.refundedAmount;
    }
    /**
     * @param { Amount } refundedAmount
     */
    setRefundedAmount(refundedAmount) {
        this.refundedAmount = ModelHelper.validateObject(refundedAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withRefundedAmount(val) {
        this.setRefundedAmount(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getRemainingRefundableAmount() {
        return this.remainingRefundableAmount;
    }
    /**
     * @param { Amount } remainingRefundableAmount
     */
    setRemainingRefundableAmount(remainingRefundableAmount) {
        this.remainingRefundableAmount = ModelHelper.validateObject(remainingRefundableAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withRemainingRefundableAmount(val) {
        this.setRemainingRefundableAmount(val);
        return this;
    }

    /**
     * @returns { Amount }
     */
    getPurchaseAmount() {
        return this.purchaseAmount;
    }
    /**
     * @param { Amount } purchaseAmount
     */
    setPurchaseAmount(purchaseAmount) {
        this.purchaseAmount = ModelHelper.validateObject(purchaseAmount, Amount);
    }
    /**
     * @param { Amount } val
     */
    withPurchaseAmount(val) {
        this.setPurchaseAmount(val);
        return this;
    }

    /**
     * @returns { Consumer }
     */
    getConsumer() {
        return this.consumer;
    }
    /**
     * @param { Consumer } consumer
     */
    setConsumer(consumer) {
        this.consumer = ModelHelper.validateObject(consumer, Consumer);
    }
    /**
     * @param { Consumer } val
     */
    withConsumer(val) {
        this.setConsumer(val);
        return this;
    }

    /**
     * @returns { ConsumerVerification }
     */
    getConsumerVerification() {
        return this.consumerVerification;
    }
    /**
     * @param { ConsumerVerification } consumerVerification
     */
    setConsumerVerification(consumerVerification) {
        this.consumerVerification = ModelHelper.validateObject(consumerVerification, ConsumerVerification);
    }
    /**
     * @param { ConsumerVerification } val
     */
    withConsumerVerification(val) {
        this.setConsumerVerification(val);
        return this;
    }

    /**
     * @returns { MerchantReference }
     */
    getMerchantReference() {
        return this.merchantReference;
    }
    /**
     * @param { MerchantReference } merchantReference
     */
    setMerchantReference(merchantReference) {
        this.merchantReference = ModelHelper.validateObject(merchantReference, MerchantReference);
    }
    /**
     * @param { MerchantReference } val
     */
    withMerchantReference(val) {
        this.setMerchantReference(val);
        return this;
    }

    /**
     * @returns { PaymentInformation }
     */
    getPaymentInformation() {
        return this.paymentInformation;
    }
    /**
     * @param { PaymentInformation } paymentInformation
     */
    setPaymentInformation(paymentInformation) {
        this.paymentInformation = ModelHelper.validateObject(paymentInformation, PaymentInformation);
    }
    /**
     * @param { PaymentInformation } val
     */
    withPaymentInformation(val) {
        this.setPaymentInformation(val);
        return this;
    }

    /**
     * @returns { [PaymentOption] }
     */
    getPaymentOptions() {
        return this.paymentOptions;
    }
    /**
     * @param { [PaymentOption] } paymentOptions
     */
    setPaymentOptions(paymentOptions) {
        this.paymentOptions = ModelHelper.validateArray(paymentOptions, PaymentOption);
    }
    /**
     * @param { [PaymentOption] } val
     */
    withPaymentOptions(val) {
        this.setPaymentOptions(val);
        return this;
    }
    /**
     * @param { PaymentOption } val
     */
    addPaymentOptions(val) {
        val = ModelHelper.validateObject(val, PaymentOption, "PaymentOption");
        this.paymentOptions = this.paymentOptions || [];
        this.paymentOptions.push(val);
    }

    /**
     * Performed captures.
     * @returns { [OperationInformation] }
     */
    getCaptures() {
        return this.captures;
    }
    /**
     * Performed captures.
     * @param { [OperationInformation] } captures
     */
    setCaptures(captures) {
        this.captures = ModelHelper.validateArray(captures, OperationInformation);
    }
    /**
     * Performed captures.
     * @param { [OperationInformation] } val
     */
    withCaptures(val) {
        this.setCaptures(val);
        return this;
    }
    /**
     * Performed captures.
     * @param { OperationInformation } val
     */
    addCaptures(val) {
        val = ModelHelper.validateObject(val, OperationInformation, "OperationInformation");
        this.captures = this.captures || [];
        this.captures.push(val);
    }

    /**
     * Performed refunds.
     * @returns { [OperationInformation] }
     */
    getRefunds() {
        return this.refunds;
    }
    /**
     * Performed refunds.
     * @param { [OperationInformation] } refunds
     */
    setRefunds(refunds) {
        this.refunds = ModelHelper.validateArray(refunds, OperationInformation);
    }
    /**
     * Performed refunds.
     * @param { [OperationInformation] } val
     */
    withRefunds(val) {
        this.setRefunds(val);
        return this;
    }
    /**
     * Performed refunds.
     * @param { OperationInformation } val
     */
    addRefunds(val) {
        val = ModelHelper.validateObject(val, OperationInformation, "OperationInformation");
        this.refunds = this.refunds || [];
        this.refunds.push(val);
    }

    /**
     * Static documents.
     * @returns { [Document] }
     */
    getDocuments() {
        return this.documents;
    }
    /**
     * Static documents.
     * @param { [Document] } documents
     */
    setDocuments(documents) {
        this.documents = ModelHelper.validateArray(documents, Document);
    }
    /**
     * Static documents.
     * @param { [Document] } val
     */
    withDocuments(val) {
        this.setDocuments(val);
        return this;
    }
    /**
     * Static documents.
     * @param { Document } val
     */
    addDocuments(val) {
        val = ModelHelper.validateObject(val, Document, "Document");
        this.documents = this.documents || [];
        this.documents.push(val);
    }

    /**
     * Additional information provided as a key value map.
     * @returns { {String: String} }
     */
    getMetaData() {
        return this.metaData;
    }
    /**
     * Additional information provided as a key value map.
     * @param { {String: String} } metaData
     */
    setMetaData(metaData) {
        this.metaData = ModelHelper.validateMap(metaData, "string", "string");
    }
    /**
     * Additional information provided as a key value map.
     * @param { {String: String} } val
     */
    withMetaData(val) {
        this.setMetaData(val);
        return this;
    }
    /**
     * Additional information provided as a key value map.
     * @param { String } key
     * @param { String } val
     */
    addMetaData(key, val) {
        key = ModelHelper.validatePrimitive(key, "string", "metaData[key]");
        val = ModelHelper.validatePrimitive(val, "string", "metaData[value]");
        this.metaData = this.metaData || {};
        this.metaData[key] = val;
    }

    /**
     * @returns { PurchaseInformation }
     */
    static constructFromObject(data) {
        if (!data) return undefined;
        return new PurchaseInformation()
            .withPurchaseId(ModelHelper.convertToType(data["purchaseId"], String))
            .withState(ModelHelper.convertToType(data["state"], PurchaseState))
            .withCurrency(ModelHelper.convertToType(data["currency"], Currency))
            .withAuthorizedAmount(ModelHelper.convertToType(data["authorizedAmount"], Amount))
            .withCapturedAmount(ModelHelper.convertToType(data["capturedAmount"], Amount))
            .withRemainingCaptureAmount(ModelHelper.convertToType(data["remainingCaptureAmount"], Amount))
            .withRefundedAmount(ModelHelper.convertToType(data["refundedAmount"], Amount))
            .withRemainingRefundableAmount(ModelHelper.convertToType(data["remainingRefundableAmount"], Amount))
            .withPurchaseAmount(ModelHelper.convertToType(data["purchaseAmount"], Amount))
            .withConsumer(ModelHelper.convertToType(data["consumer"], Consumer))
            .withConsumerVerification(ModelHelper.convertToType(data["consumerVerification"], ConsumerVerification))
            .withMerchantReference(ModelHelper.convertToType(data["merchantReference"], MerchantReference))
            .withPaymentInformation(ModelHelper.convertToType(data["paymentInformation"], PaymentInformation))
            .withPaymentOptions(ModelHelper.convertToType(data["paymentOptions"], [PaymentOption]))
            .withCaptures(ModelHelper.convertToType(data["captures"], [OperationInformation]))
            .withRefunds(ModelHelper.convertToType(data["refunds"], [OperationInformation]))
            .withDocuments(ModelHelper.convertToType(data["documents"], [Document]))
            .withMetaData(ModelHelper.convertToType(data["metaData"], { String: String }));
    }
}

module.exports = PurchaseInformation;

File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/PurchaseInformation.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

import java.util.ArrayList;
import java.util.List;
import java.util.ArrayList;
import java.util.List;
import java.util.ArrayList;
import java.util.List;
import java.util.ArrayList;
import java.util.List;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

/**
 * Describes the current state of a purchase.
 */
public class PurchaseInformation {

    private String purchaseId;
    private PurchaseState state;
    private Currency currency;
    private Amount authorizedAmount;
    private Amount capturedAmount;
    private Amount remainingCaptureAmount;
    private Amount refundedAmount;
    private Amount remainingRefundableAmount;
    private Amount purchaseAmount;
    private Consumer consumer;
    private ConsumerVerification consumerVerification;
    private MerchantReference merchantReference;
    private PaymentInformation paymentInformation;
    private List<PaymentOption> paymentOptions;
    private List<OperationInformation> captures;
    private List<OperationInformation> refunds;
    private List<Document> documents;
    private Map<String, String> metaData;

    /**
     * @return purchaseId
     */
    public String getPurchaseId() {
        return purchaseId;
    }
    public void setPurchaseId(String purchaseId) {
        this.purchaseId = purchaseId;
    }
    public PurchaseInformation withPurchaseId(String purchaseId) {
        this.purchaseId = purchaseId;
        return this;
    }

    /**
     * @return state
     */
    public PurchaseState getState() {
        return state;
    }
    public void setState(PurchaseState state) {
        this.state = state;
    }
    public PurchaseInformation withState(PurchaseState state) {
        this.state = state;
        return this;
    }

    /**
     * @return currency
     */
    public Currency getCurrency() {
        return currency;
    }
    public void setCurrency(Currency currency) {
        this.currency = currency;
    }
    public PurchaseInformation withCurrency(Currency currency) {
        this.currency = currency;
        return this;
    }

    /**
     * @return authorizedAmount
     */
    public Amount getAuthorizedAmount() {
        return authorizedAmount;
    }
    public void setAuthorizedAmount(Amount authorizedAmount) {
        this.authorizedAmount = authorizedAmount;
    }
    public PurchaseInformation withAuthorizedAmount(Amount authorizedAmount) {
        this.authorizedAmount = authorizedAmount;
        return this;
    }

    /**
     * @return capturedAmount
     */
    public Amount getCapturedAmount() {
        return capturedAmount;
    }
    public void setCapturedAmount(Amount capturedAmount) {
        this.capturedAmount = capturedAmount;
    }
    public PurchaseInformation withCapturedAmount(Amount capturedAmount) {
        this.capturedAmount = capturedAmount;
        return this;
    }

    /**
     * @return remainingCaptureAmount
     */
    public Amount getRemainingCaptureAmount() {
        return remainingCaptureAmount;
    }
    public void setRemainingCaptureAmount(Amount remainingCaptureAmount) {
        this.remainingCaptureAmount = remainingCaptureAmount;
    }
    public PurchaseInformation withRemainingCaptureAmount(Amount remainingCaptureAmount) {
        this.remainingCaptureAmount = remainingCaptureAmount;
        return this;
    }

    /**
     * @return refundedAmount
     */
    public Amount getRefundedAmount() {
        return refundedAmount;
    }
    public void setRefundedAmount(Amount refundedAmount) {
        this.refundedAmount = refundedAmount;
    }
    public PurchaseInformation withRefundedAmount(Amount refundedAmount) {
        this.refundedAmount = refundedAmount;
        return this;
    }

    /**
     * @return remainingRefundableAmount
     */
    public Amount getRemainingRefundableAmount() {
        return remainingRefundableAmount;
    }
    public void setRemainingRefundableAmount(Amount remainingRefundableAmount) {
        this.remainingRefundableAmount = remainingRefundableAmount;
    }
    public PurchaseInformation withRemainingRefundableAmount(Amount remainingRefundableAmount) {
        this.remainingRefundableAmount = remainingRefundableAmount;
        return this;
    }

    /**
     * @return purchaseAmount
     */
    public Amount getPurchaseAmount() {
        return purchaseAmount;
    }
    public void setPurchaseAmount(Amount purchaseAmount) {
        this.purchaseAmount = purchaseAmount;
    }
    public PurchaseInformation withPurchaseAmount(Amount purchaseAmount) {
        this.purchaseAmount = purchaseAmount;
        return this;
    }

    /**
     * @return consumer
     */
    public Consumer getConsumer() {
        return consumer;
    }
    public void setConsumer(Consumer consumer) {
        this.consumer = consumer;
    }
    public PurchaseInformation withConsumer(Consumer consumer) {
        this.consumer = consumer;
        return this;
    }

    /**
     * @return consumerVerification
     */
    public ConsumerVerification getConsumerVerification() {
        return consumerVerification;
    }
    public void setConsumerVerification(ConsumerVerification consumerVerification) {
        this.consumerVerification = consumerVerification;
    }
    public PurchaseInformation withConsumerVerification(ConsumerVerification consumerVerification) {
        this.consumerVerification = consumerVerification;
        return this;
    }

    /**
     * @return merchantReference
     */
    public MerchantReference getMerchantReference() {
        return merchantReference;
    }
    public void setMerchantReference(MerchantReference merchantReference) {
        this.merchantReference = merchantReference;
    }
    public PurchaseInformation withMerchantReference(MerchantReference merchantReference) {
        this.merchantReference = merchantReference;
        return this;
    }

    /**
     * @return paymentInformation
     */
    public PaymentInformation getPaymentInformation() {
        return paymentInformation;
    }
    public void setPaymentInformation(PaymentInformation paymentInformation) {
        this.paymentInformation = paymentInformation;
    }
    public PurchaseInformation withPaymentInformation(PaymentInformation paymentInformation) {
        this.paymentInformation = paymentInformation;
        return this;
    }

    /**
     * @return paymentOptions
     */
    public List<PaymentOption> getPaymentOptions() {
        return paymentOptions;
    }
    public void setPaymentOptions(List<PaymentOption> paymentOptions) {
        this.paymentOptions = paymentOptions;
    }
    public PurchaseInformation withPaymentOptions(List<PaymentOption> paymentOptions) {
        this.paymentOptions = paymentOptions;
        return this;
    }
    public PurchaseInformation addPaymentOptions(PaymentOption value) {
        if (this.paymentOptions == null) {
            this.paymentOptions = new ArrayList<>();
        }
        this.paymentOptions.add(value);
        return this;
    }

    /**
     * Performed captures.
     *
     * @return captures
     */
    public List<OperationInformation> getCaptures() {
        return captures;
    }
    public void setCaptures(List<OperationInformation> captures) {
        this.captures = captures;
    }
    public PurchaseInformation withCaptures(List<OperationInformation> captures) {
        this.captures = captures;
        return this;
    }
    public PurchaseInformation addCaptures(OperationInformation value) {
        if (this.captures == null) {
            this.captures = new ArrayList<>();
        }
        this.captures.add(value);
        return this;
    }

    /**
     * Performed refunds.
     *
     * @return refunds
     */
    public List<OperationInformation> getRefunds() {
        return refunds;
    }
    public void setRefunds(List<OperationInformation> refunds) {
        this.refunds = refunds;
    }
    public PurchaseInformation withRefunds(List<OperationInformation> refunds) {
        this.refunds = refunds;
        return this;
    }
    public PurchaseInformation addRefunds(OperationInformation value) {
        if (this.refunds == null) {
            this.refunds = new ArrayList<>();
        }
        this.refunds.add(value);
        return this;
    }

    /**
     * Static documents.
     *
     * @return documents
     */
    public List<Document> getDocuments() {
        return documents;
    }
    public void setDocuments(List<Document> documents) {
        this.documents = documents;
    }
    public PurchaseInformation withDocuments(List<Document> documents) {
        this.documents = documents;
        return this;
    }
    public PurchaseInformation addDocuments(Document value) {
        if (this.documents == null) {
            this.documents = new ArrayList<>();
        }
        this.documents.add(value);
        return this;
    }

    /**
     * Additional information provided as a key value map.
     *
     * @return metaData
     */
    public Map<String, String> getMetaData() {
        return metaData;
    }
    public void setMetaData(Map<String, String> metaData) {
        this.metaData = metaData;
    }
    public PurchaseInformation withMetaData(Map<String, String> metaData) {
        this.metaData = metaData;
        return this;
    }
    public PurchaseInformation addMetaData(String key, String value) {
        if (this.metaData == null) {
            this.metaData = new HashMap<>();
        }
        this.metaData.put(key, value);
        return this;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        PurchaseInformation purchaseInformation = (PurchaseInformation) o;
        return Objects.equals(this.purchaseId, purchaseInformation.purchaseId)
                && Objects.equals(this.state, purchaseInformation.state)
                && Objects.equals(this.currency, purchaseInformation.currency)
                && Objects.equals(this.authorizedAmount, purchaseInformation.authorizedAmount)
                && Objects.equals(this.capturedAmount, purchaseInformation.capturedAmount)
                && Objects.equals(this.remainingCaptureAmount, purchaseInformation.remainingCaptureAmount)
                && Objects.equals(this.refundedAmount, purchaseInformation.refundedAmount)
                && Objects.equals(this.remainingRefundableAmount, purchaseInformation.remainingRefundableAmount)
                && Objects.equals(this.purchaseAmount, purchaseInformation.purchaseAmount)
                && Objects.equals(this.consumer, purchaseInformation.consumer)
                && Objects.equals(this.consumerVerification, purchaseInformation.consumerVerification)
                && Objects.equals(this.merchantReference, purchaseInformation.merchantReference)
                && Objects.equals(this.paymentInformation, purchaseInformation.paymentInformation)
                && Objects.equals(this.paymentOptions, purchaseInformation.paymentOptions)
                && Objects.equals(this.captures, purchaseInformation.captures)
                && Objects.equals(this.refunds, purchaseInformation.refunds)
                && Objects.equals(this.documents, purchaseInformation.documents)
                && Objects.equals(this.metaData, purchaseInformation.metaData);
    }

    @Override
    public int hashCode() {
        return Objects.hash(purchaseId, state, currency, authorizedAmount, capturedAmount, remainingCaptureAmount, refundedAmount, remainingRefundableAmount, purchaseAmount, consumer, consumerVerification, merchantReference, paymentInformation, paymentOptions, captures, refunds, documents, metaData);
    }

    @Override
    public String toString() {
        return "class PurchaseInformation {\n"
                + "        purchaseId: " + toIndentedString(purchaseId) + "\n"
                + "        state: " + toIndentedString(state) + "\n"
                + "        currency: " + toIndentedString(currency) + "\n"
                + "        authorizedAmount: " + toIndentedString(authorizedAmount) + "\n"
                + "        capturedAmount: " + toIndentedString(capturedAmount) + "\n"
                + "        remainingCaptureAmount: " + toIndentedString(remainingCaptureAmount) + "\n"
                + "        refundedAmount: " + toIndentedString(refundedAmount) + "\n"
                + "        remainingRefundableAmount: " + toIndentedString(remainingRefundableAmount) + "\n"
                + "        purchaseAmount: " + toIndentedString(purchaseAmount) + "\n"
                + "        consumer: " + toIndentedString(consumer) + "\n"
                + "        consumerVerification: " + toIndentedString(consumerVerification) + "\n"
                + "        merchantReference: " + toIndentedString(merchantReference) + "\n"
                + "        paymentInformation: " + toIndentedString(paymentInformation) + "\n"
                + "        paymentOptions: " + toIndentedString(paymentOptions) + "\n"
                + "        captures: " + toIndentedString(captures) + "\n"
                + "        refunds: " + toIndentedString(refunds) + "\n"
                + "        documents: " + toIndentedString(documents) + "\n"
                + "        metaData: " + toIndentedString(metaData) + "\n"
                + "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        return o == null
                ? "null"
                : o.toString().replace("\n", "\n        ");
    }
}

